pennylane/ops/op_math/prod.py

```diff
@@ -421,6 +432,177 @@ def _swappable_ops(op1, op2, wire_map: dict = None) -> bool:
+        there was another Pauli operator acting on the same wire, the two operators are grouped
+        together using the ``self._pauli_mult`` dictionary.
+
+        Args:
+            factor (Operator): Factor to be added.
+            wires (List[int]): Factor wires. This argument is added to avoid calling
+                ``factor.wires`` several times.
+        """
+        wire = wires[0]
+        op2_name = factor.name
+        old_coeff, old_word = self._pauli_factors.get(wire, (1, "Identity"))
+        coeff, new_word = self._pauli_mult[old_word][op2_name]
+        self._pauli_factors[wire] = old_coeff * coeff, new_word
+
+    def _add_non_pauli_factor(self, factor: Operator, wires: List[int]):
+        """Adds the given non-Pauli factor to the temporary ``self._non_pauli_factors`` dictionary.
+        If there alerady exists an identical operator in the dictionary, the two are grouped
+        together.
+
+        If there isn't an identical operator in the dictionary, all non Pauli factors that act on
+        the same wires are removed and added to the ``self._factors`` tuple.
+
+        Args:
+            factor (Operator): Factor to be added.
+            wires (List[int]): Factor wires. This argument is added to avoid calling
+                ``factor.wires`` several times.
+        """
+        if isinstance(factor, Pow):
+            exponent = factor.z
+            factor = factor.base
+        else:
+            exponent = 1
+        op_hash = factor.hash
+        old_hash, old_exponent, old_op = self._non_pauli_factors.get(wires, [None, None, None])
+        if isinstance(old_op, (qml.RX, qml.RY, qml.RZ)) and factor.name == old_op.name:
+            self._non_pauli_factors[wires] = [
+                op_hash,
+                old_exponent,
+                factor.__class__(factor.data[0] + old_op.data[0], wires).simplify(),
+            ]
+        elif op_hash == old_hash:
+            self._non_pauli_factors[wires][1] += exponent
+        else:
+            self._remove_non_pauli_factors(wires=wires)
+            self._non_pauli_factors[wires] = [op_hash, copy(exponent), factor]
+
+    def _remove_non_pauli_factors(self, wires: List[int]):
+        """Remove all factors from the ``self._non_pauli_factors`` dictionary that act on the given
+        wires and add them to the ``self._factors`` tuple.
+
+        Args:
+            wires (List[int]): Wires of the operators to be removed.
+        """
+        if not self._non_pauli_factors:
+            return
+        for wire in wires:
+            for key, (_, exponent, op) in list(self._non_pauli_factors.items()):
+                if wire in key:
+                    self._non_pauli_factors.pop(key)
+                    if exponent == 0:
+                        continue
+                    # TODO: Should we create a qml.pow function that calls op.pow() if possible?
+                    op = Pow(base=op, z=exponent).simplify() if exponent != 1 else op
```

Comment:
I think we can do this, don't need the else 
```suggestion
                    if exponent !=1: op = Pow(base=op, z=exponent).simplify()
```